addons/test/misra/misra-test.c

```diff
@@ -233,6 +233,17 @@ void misra_7_3() {
   long double misra_7_3_e = 7.3l; //7.3
   }
 
+void misra_7_4_const_call(int a, const char* b) { } // 2.7
+void misra_7_4_call(int a, char* b) { } // 2.7
+
+void misra_7_4()
+{
+   const char *a = "string_a"; 
+   char *b = "string_b";  // 7.4
```

Comment:
hmm.. I have the feeling we should modify the Cppcheck AST for this code:

    char data[] = "abcdef";

So that the astOperand1 for `=` is `data`. That will mean the misra addon needs to check if it's assigned to a pointer or not.

Reply:
Ok, I can look into this.